examples/workflows/experimental/script_runner_io_dynamic_fanout.py

```diff
@@ -0,0 +1,38 @@
+"""
```

Comment:
I don't think we need to add these two examples - they are not demonstrating something unique - script annotations/pydantic IO have their own examples, and the syntax for `with_param` is more simply shown in the loops examples, where the syntax in the DAG construction is the same.

Reply:
Fair, I added them to exercise the code rather than as examples per se. Is there a better place to put this kind of end-to-end YAML output test?

Reply:
You could do a roundtrip `to_dict`/`from_dict` like in the `test_script_annotations` tests https://github.com/argoproj-labs/hera/blob/a08f744cd9d7ca30091e7ab51e80bd18e3116010/tests/test_script_annotations.py#L373-L374